src/mlpack/core/util/output_param_impl.hpp

```diff
@@ -48,9 +48,16 @@ void OutputParamImpl(
 {
   const T& output = *boost::any_cast<T>(&data.mappedValue);
   const std::string& filename = *boost::any_cast<std::string>(&data.value);
-
+  
   if (output.n_elem > 0 && filename != "")
-    data::Save(filename, output, false, !data.noTranspose);
+  {
+    //call the correct data::Save()
+    if (data.tname.find("Col") != std::string::npos || data.tname.find("Row") != std::string::npos)
```

Comment:
This is a clever way to do it, but unfortunately there's no guarantee that the typename (as given by RTTI) has the class name in it.  On gcc and clang that will probably always work, but it won't be portable to every possible compiler. :)

I think, maybe a better way to do this, might be to use the (undocumented) `arma::is_Row<T>` and `arma::is_Col<T>` structs:

```
if (arma::is_Row<T>::value || arma::is_Col<T>::value)
```

Technically these structs are not documented by Armadillo and we run the risk that future versions may remove or change the names, but in my opinion it doesn't make much sense to reinvent the wheel by making our own identical structs, and if Armadillo changes we can backport or otherwise handle it when that happens.